pkg/openstack/loadbalancer.go

```diff
@@ -2256,11 +2291,31 @@ func (lbaas *LbaasV2) ensureAndUpdateOctaviaSecurityGroup(clusterName string, ap
 				"failed to apply security rule for port %d, %w",
 				port.NodePort, err)
 		}
+		wantedRuleIDs = append(wantedRuleIDs, id)
+	}
 
-		if err := applyNodeSecurityGroupIDForLB(lbaas.compute, lbaas.network, nodes, lbSecGroupID); err != nil {
-			return err
+	// delete unneeded rules
+	existingRules, err := getSecurityGroupRules(lbaas.network, rules.ListOpts{SecGroupID: lbSecGroupID})
+	if err != nil && !cpoerrors.IsNotFound(err) {
+		return fmt.Errorf(
+			"failed to find security group rules in %s: %v", lbSecGroupID, err)
+	}
+	for _, existingRule := range existingRules {
+		needed := false
+		for _, wantedRuleID := range wantedRuleIDs {
+			if wantedRuleID == existingRule.ID {
+				needed = true
+				break
+			}
+		}
+		if !needed {
```

Comment:
can we at least log.v5 or v6 to record this deletion ? sometimes this will be helpful as well for debug..